src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -23,23 +24,25 @@ internal DefaultArrayPool(int maxLength, int arraysPerBucket)
             // Our bucketing algorithm has a minimum length of 16
             if (maxLength < MinimiumArraySize)
                 maxLength = MinimiumArraySize;
-            
+
+            int id = GetHashCode();
```

Comment:
Nit: Consider adding a property like:
```C#
private int PoolId { get { return GetHashCode(); } }
```
so that call sites can use ```PoolId``` instead of ```GetHashCode()```.  That'll make it clear why GetHashCode() is being called / what it's being used for, and also allow changing it more easily in the future.